src/databricks/labs/ucx/installer/logs.py

```diff
@@ -0,0 +1,77 @@
+import dataclasses
+import logging
+import re
+from collections.abc import Iterator
+from pathlib import Path
+
+from databricks.labs.lsql.backends import SqlBackend
+
+from databricks.labs.ucx.framework.crawlers import CrawlerBase
+
+logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class LogRecord:
+    """A subset from logging.LogRecord.
+
+    Sources
+    -------
+    https://docs.python.org/3/library/logging.html#logging.LogRecord
+    """
+
+    level: int
+    msg: str
+
+
+def parse_logs(*log_paths: Path) -> Iterator[LogRecord]:
+    log_format = r"\d+:\d+:\d+\s(\w+)\s\[.+\]\s\{\w+\}\s(.+)"
+    pattern = re.compile(log_format)
+
+    for log_path in log_paths:
+        if not log_path.is_file():
+            logger.info("Log file does not exists: {%s}", log_path)
+            continue
+        with log_path.open("r") as f:
```

Comment:
separate parsing logic from file reading logic. to do https://github.com/databrickslabs/ucx/issues/1186 we'll only be able to use `ws.workspace.download(...)`, not `pathlib.Path`